packages/lib/src/alert/Alert.tsx

```diff
@@ -149,11 +149,15 @@ export default function DxcAlert({
   };
   const handleOnClose = () => {
     messages[currentIndex]?.onClose?.();
-    mode !== "modal" && setMessages((prevMessages) => prevMessages.filter((_, index) => index !== currentIndex));
+    if (mode !== "modal") {
```

Comment:
In cases like this, a linter rule (`react-hooks/exhaustive-deps`) was throwing a warning due to calling a `set` directly in a conditional expression because it can lead to potential issues with how the state is updated and dependencies are managed. It is generally more encouraged to call it inside of an `if` statement. We can talk about this further anyway.